instance/models/instance.py

```diff
@@ -119,11 +123,13 @@ class Instance(ValidateModelMixin, models.Model):
     # in a query, e.g. to do .select_related('ref_set')
     ref_set = GenericRelation(InstanceReference, content_type_field='instance_type', object_id_field='instance_id')
     openstack_region = models.CharField(
+        # TODO: Move this to OpenedXInstance
         max_length=16,
         blank=False,
         default=default_setting('OPENSTACK_REGION'),
     )
     tags = models.ManyToManyField(
+        # TODO: Move this to InstanceReference (common fields should be in non-abstract tables where possible)
```

Comment:
@bradenmacdonald I can follow some of the arguments, but not all of them.  We would not need to repeat the field on different instance types, since they will all inherit it from `Instance`, so I don't see a DRY violation.  Querying tags across different instance types feels rather contrived to me; I don't think we will ever use tags this way.  And regarding the indirection, I was mostly worried about the reverse relation:  Retrieving all instances with a given tag.  Currently, that's simply

    tag.openedxinstance_set.all()

If we move the field to the instance reference, the only reasonable way to do this is this

    OpenEdXInstance.objects.filter(ref_set__tags=tag)

That's not too bad, but it will actually become less efficient due to the additional join, and this is in my opinion the most common use case.

That said, I'm not really opposed to moving the tags to the instance reference.  I just don't see strong arguments either way.